pkg/agent/controller/lhserviceexport.go

```diff
@@ -0,0 +1,74 @@
+package controller
+
+import (
+	"github.com/submariner-io/admiral/pkg/syncer"
+	"github.com/submariner-io/admiral/pkg/syncer/broker"
+	lighthousev2a1 "github.com/submariner-io/lighthouse/pkg/apis/lighthouse.submariner.io/v2alpha1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/klog"
+	mcsv1a1 "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+)
+
+func newLHServiceExportController(localClient dynamic.Interface, restMapper meta.RESTMapper,
+	scheme *runtime.Scheme) (*LHServiceExportController, error) {
+	serviceExportController := LHServiceExportController{
+		localClient: localClient,
+	}
+
+	lhServiceExportSyncer, err := syncer.NewResourceSyncer(&syncer.ResourceSyncerConfig{
+		Name:             "lhServiceExport -> mcsServiceExport",
+		SourceClient:     localClient,
+		SourceNamespace:  metav1.NamespaceAll,
+		Direction:        syncer.LocalToRemote,
+		RestMapper:       restMapper,
+		Federator:        broker.NewFederator(localClient, restMapper, metav1.NamespaceAll, ""),
+		ResourceType:     &lighthousev2a1.ServiceExport{},
+		Transform:        serviceExportController.LHtoMCSServiceExport,
+		Scheme:           scheme,
+		OnSuccessfulSync: serviceExportController.deleteLHServiceExport,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	serviceExportController.lhServiceExportSyncer = lhServiceExportSyncer
+
+	return &serviceExportController, nil
+}
+
+func (c *LHServiceExportController) start(stopCh <-chan struct{}) error {
+	if err := c.lhServiceExportSyncer.Start(stopCh); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *LHServiceExportController) LHtoMCSServiceExport(obj runtime.Object, op syncer.Operation) (runtime.Object, bool) {
+	serviceExportCreated := obj.(*lighthousev2a1.ServiceExport)
```

Comment:
We only want to handle creates here. We delete the LH ServiceExport which I think would re-trigger this syncer to delete the MCS ServiceExport.  I think it would be good to add a unit test to cover this.

```
if op != syncer.Create {
    return nil, false
}
```

Reply:
Done. , added UT.